api/src/main/java/io/grpc/ServerCall.java

```diff
@@ -163,6 +163,12 @@ public boolean isReady() {
    * <p>Since {@link Metadata} is not thread-safe, the caller must not access (read or write) {@code
    * trailers} after this point.
    *
+   * <p>Implementation note: This method is called after the server completed the processing of the
+   * request (service implementation), but before the response status is sent to the client, so the
```

Comment:
I like that, but I'd really hope to avoid "implementation note." This is more how the API is designed versus how things just so happen to be implemented.

> This method implies the caller completed processing the RPC, but it does not imply the RPC is complete. The call implementation will need additional time to complete the RPC and during this time the client is still able to cancel the request or a network error might cause the RPC to fail. If you wish to know...

It could maybe use some more wordsmithing, but is that better?

Did you want to include anything along the lines of "this method is not necessary when Listener.onCancel() is called?"